src/cmd.c

```diff
@@ -2085,6 +2085,53 @@ static void set_nvm(atci_param_t *param) {
     }
 }
 
+#if MKR1310 == 1
+// The MKR1310 is using the same wires for SPI and UART
+// To be able to use the embedded SPI Flash, we need to switch UART Off
+// As uart is used for waking the device up when going into sleep mode
+// we will disable the sleep mode too during this period of time
+// Uart will be resumed by setting GPIO B12 to LOW (it has a pull-up on the line)
+static uint8_t __prev_sleep;
+static uint8_t __uart_disable = 0;
+static void disable_uart(atci_param_t *param) {
+    // simpler to add everything in a single place, 
+    // make sure these lines are INPUT to not conflict with external signals from SPI
+    GPIO_InitTypeDef cfg = {
+        .Mode = GPIO_MODE_INPUT,
+        .Pull = GPIO_NOPULL
+    };
+    gpio_init(GPIOB, GPIO_PIN_12, &cfg);
+    gpio_init(GPIOB, GPIO_PIN_13, &cfg);
+    gpio_init(GPIOB, GPIO_PIN_14, &cfg);
+    gpio_init(GPIOB, GPIO_PIN_15, &cfg);
+
+    // No need to deactivate uart if the wakeup signal is true
+    int v = gpio_read(GPIOB, GPIO_PIN_12);
+    if ( v == 0 ) abort(ERR_PARAM);
+    OK_();
+
+    lpuart_disable();
+    __prev_sleep = sysconf.sleep;
+    __uart_disable = 1;
+    sysconf.sleep = 0; // prevent the device to go deep sleep and never waked up due to serial line deactivation
+}
+
+// this is monitoring the PB12 line and restart the UART when to comes low
+void process_uart_wakeup(void) {
+    // nothing to do
+    if ( __uart_disable == 0 ) return;
+
+    int v = gpio_read(GPIOB, GPIO_PIN_12);
+    if ( v == 0 ) {
+        // restart is requested
+        lpuart_enable();
+        sysconf.sleep = __prev_sleep;
+        __uart_disable = 0;
+        OK_();
+    }
+}
+
+#endif
```

Comment:
This is really specific to MKR1310 because the same wires are used for ABZ UART and Flash SPI. As a consequence you can't use the onboarded flash when the UART is activ.
I did remove sleep when uart is disabled because only the UART wakes the ABZ up when going to sleep, so if UART is down .. that was a lower impact on the code than changing the sleep functions to wakeup on a GPIO. You should use the flash on short duration so it have no impact on overall energy consumption. 
This is really a bypass for a bug of coneption on MKR1310

Reply:
I checked the MKRWAN1310 schematic and I see what the problem is now.

Do you prefer your solution (disabling LPUART temporarily), or should we attempt something else? Would it make sense, for example, to design a SPI-based interface as an alternative to the AT command interface? I am not that worried about the amount of additional code and an alternative to the ATCI would seem generally useful.

If you don't mind, please spin this feature off into a separate PR and we can discuss it there.

Reply:
I should have made different branches for the PR but unfortunatelly thay are on master, so I can dissociate them. But I think you can accept to merge commit by commit and the commit are for the different features. So you can delay the commit related to the UART / SPI.
I don't think making a SPI interface is a good idea as it changes the Arduino lib, even if rewritting it would be a good idea as it is badly written, It will not be adopted by the community, they won't want to port their code to a new lib if they use the default one. So my commit is the best solution I found against many worst them. I think we should keep that one.

Reply:
Oops. It looks like I edited your comment instead of posting a reply. Sorry! Github UI mixup, I guess.

Reply:
go it

Reply:
using GPIO wake up would be good as this is also the way I use to restore the lpuart.
I prefer you make this modif as I know (for having implemented a lot around) that is is sensitive to touch the low-power code and can have side impact on functions I did not touched in my modifications